control/controlclient/auto.go

```diff
@@ -118,8 +77,9 @@ func (g *LoginGoal) sendLogoutError(err error) {
 	}
 }
 
-// Client connects to a tailcontrol server for a node.
```

Comment:
Let's keep this as Client but rename the interface instead.

And unexport the Direct type.

Reply:
I'm not super impressed with the naming, but I think the type outsiders are expected to use (the interface) should be named Client. I'll rename AutoClient back to Auto since controlclient.AutoClient is redundant. (So is controlclient.Client, alas.)

I'd love to unexport Direct, but I can't because of the silly external tests.